healthcheck-sofa-boot-starter/src/main/java/com/alipay/sofa/healthcheck/service/SofaBootReadinessCheckMvcEndpoint.java

```diff
@@ -36,10 +37,12 @@
 /**
  * @author khotyn
  */
+@ConfigurationProperties(prefix = "endpoints.sofaboot.health.readiness")
```

Comment:
这不能统一属性配置吧 https://docs.spring.io/spring-boot/docs/current/reference/html/production-ready-endpoints.html#production-ready-endpoints-enabling-endpoints 每个endpoint 应该都需要提供用户配置，disable/enable

Reply:
> 每个endpoint 应该都需要提供用户配置，disable/enable 

这个是可以的。由于我们的规范前缀和 Spring Boot 不一样，要改为我们的规范。

同时，也不是每一个 Endpoint 都要配置的，注意我们的代码，在 `SofaBootReadinessCheckMvcEndpoint.java` 有如下的代码：

```java
public Object invoke(Principal principal) {
        if (!getDelegate().isEnabled()) {
            // Shouldn't happen because the request mapping should not be registered
            return getDisabledResponse();
        }
```
而这个 Deletage 是 `com.alipay.sofa.healthcheck.service.SofaBootReadinessCheckEndpoint`，所以我们只需要在 `com.alipay.sofa.healthcheck.service.SofaBootReadinessCheckEndpoint` 支持 disable/enable 即可，当然我们默认应该是 enable 的。

Reply:
@QilongZhang 最后再补充一下测试用例吧，针对这个的测试用例，也补充下。

Reply:
不太赞成 Delegate 判断enable这种做法，这段代码一开始看到了的，其实这段代码已经可以删掉了，因为已经在
```
    @Bean
    @ConditionalOnBean(SofaBootReadinessCheckEndpoint.class)
    @ConditionalOnEnabledEndpoint(READINESS_CHECK_ENDPOINT_NAME)
    public SofaBootReadinessCheckMvcEndpoint sofaBootReadinessCheckMvcEndpoint(SofaBootReadinessCheckEndpoint delegate) {
        return new SofaBootReadinessCheckMvcEndpoint(delegate);
    }
```
按照这次pr的写法，当SofaBootReadinessCheckEndpoint被disable掉之后，这个bean就不会创建了。 以前的缺点是，这个bean同样是会创建出来，日志也会打印相应的url，不认为是一种好的写法。

Reply:
雪总，我的建议是，既然深度兼容spring boot， sofa boot本身不应该制定太多的规范，如果spring boot 有一套通用的写法习惯，应该是向它靠拢的。

Reply:
我们的配置规范是 `com.alipay.sofa.${component}`，有一个统一的配置规范这里我们的开源组件都按照这个规范。有产品标识也有组件标识，这个就不要再单独搞了，要统一我们的规范，[参考这里样例](https://github.com/alipay/sofa-boot/wiki/HealthCheck#readiness-check-%E9%85%8D%E7%BD%AE%E9%A1%B9)

Reply:
@QilongZhang 配置上 SOFA 还是用自己的整套命名规则体系吧。

Reply:
@khotyn @guanchao-yang  我不是说命名规范哈。 是说mvc endpoint 和 delegate 应该共用一个变量控制，而不是只对delegate提供disable/enable,  mvc endpoint 依然注册 request mapping，在内部通过delegate判断。这个不需要改命名规范，变量前缀还是用 `com.alipay.sofa.` 即可。